peps/pep-0764.rst

```diff
@@ -0,0 +1,334 @@
+
+    def get_movie() -> dict[{'title': str}]: ...
+
+While this would avoid having to import :class:`~typing.TypedDict` from
+:mod:`typing`, this solution has several downsides:
+
+* For type checkers, :class:`dict` is a regular class with two type variables.
+  Allowing :class:`dict` to be parametrized with a single type argument would
+  require special casing from type checkers, as there is no way to express
+  parametrization overloads. On the other hand, :class:`~typing.TypedDict` is
+  already a :term:`special form <typing:special form>`.
+
+* If future work extends what inlined typed dictionaries can do, we don't have
+  to worry about impact of sharing the symbol with :class:`dict`.
+
+Using a simple dictionary
+-------------------------
+
+Instead of subscripting the :class:`~typing.TypedDict` class, a plain
+dictionary could be used as an annotation::
+
+    def get_movie() -> {'title': str}: ...
+
+However, :pep:`584` added union operators on dictionaries and :pep:`604`
+introduced :ref:`union types <python:types-union>`. Both features make use of
+the :ref:`bitwise or (|) <python:bitwise>` operator, making the following use
+cases incompatible, especially for runtime introspection::
+
+    # Dictionaries are merged:
+    def fn() -> {'a': int} | {'b': str}: ...
+
+    # Raises a type error at runtime:
+    def fn() -> {'a': int} | int: ...
+
+Open Issues
+===========
+
+Subclassing an inlined typed dictionary
+---------------------------------------
+
+Should we allow the following?::
+
+    from typing import TypedDict
+
+    InlinedTD = TypedDict[{'a': int}]
+
+
+    class SubTD(InlinedTD):
+        pass
+
+What about defining an inlined typed dictionay extending another typed
+dictionary?::
+
+    InlinedBase = TypedDict[{'a': int}]
+
+    Inlined = TypedDict[InlinedBase, {'b': int}]
+
+Using ``typing.Dict`` with a single argument
+--------------------------------------------
+
+While using :class:`dict` isn't ideal, we could make use of
+:class:`typing.Dict` with a single argument::
+
+    def get_movie() -> Dict[{'title': str}]: ...
+
+It is less verbose, doesn't have the baggage of :class:`dict`, and is
+already defined as some kind of special form.
+
+However, it is currently marked as deprecated (although not scheduled for
+removal), so it might be confusing to undeprecate it.
+
+This would also set a precedent on typing constructs being parametrizable
+with a different number of type arguments.
+
+Should inlined typed dictionaries be proper classes?
+----------------------------------------------------
+
+The PEP currently defines inlined typed dictionaries as type objects, to be in
+line with the existing syntaxes. To work around the fact that they don't have
+a name, their :attr:`~type.__name__` attribute is set to an empty string.
+
+This is somewhat arbitrary, and an alternative name could be used as well
+(e.g. ``'<TypedDict>'``).
+
+Alternatively, inlined typed dictionaries could be defined as instances of a
```

Comment:
This might actually be better. Runtime introspection tools would likely already need some updates, or they'll output really confusing diagnostics for a TypedDict with an empty name. (Though I just tried and Pydantic doesn't seem to output the TypedDict name in its error message, so maybe it would be fine for you.)

Reply:
> as the sponsor could you confirm you're also OK with merging the PEP as it stands?

Yes, I'm good with merging the latest draft.